MQTTPacket/test/CMakeLists.txt

```diff
@@ -47,6 +47,9 @@ ADD_EXECUTABLE(
 #  warning: ‘ftime’ is deprecated [-Wdeprecated-declarations]
 target_compile_options(test15 PRIVATE -Wno-deprecated-declarations)
 
+# To allow existing MQTTv3 applications using the MQTTv5 library, MQTTV5 structures must be used.
+target_compile_definitions(test15 PRIVATE MQTTV5)
```

Comment:
This is something that I'm not very happy about but I'm not sure about the backward compatibility scenario given that MQTTv5 has a few different behaviors.
Proposed V5 client in https://github.com/eclipse/paho.mqtt.embedded-c/pull/244 will likely complicate things even more as more structs will be modified.

Reply:
Agreed: I will update this PR to remove access to v3 APIs when `MQTTV5Packet.h` is used.